engine/access/rpc/backend/backend_accounts.go

```
@@ -81,29 +81,26 @@ func (b *backendAccounts) getAccountAtBlockID(
 		BlockId: blockID[:],
 	}
 
-	var exeRes *execproto.GetAccountAtBlockIDResponse
-	var err error
-
-	if b.staticExecutionRPC != nil {
+	execNodes, err := executionNodesForBlockID(blockID, b.executionReceipts, b.state)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get account from the execution node: %v", err)
+	}
 
-		exeRes, err = b.staticExecutionRPC.GetAccountAtBlockID(ctx, &exeReq)
-		if err != nil {
-			convertedErr := getAccountError(err)
-			return nil, convertedErr
+	var exeRes *execproto.GetAccountAtBlockIDResponse
+	if len(execNodes) == 0 {
+		if b.staticExecutionRPC != nil {
+			exeRes, err = b.staticExecutionRPC.GetAccountAtBlockID(ctx, &exeReq)
+			if err != nil {
+				convertedErr := getAccountError(err)
+				return nil, convertedErr
+			}
 		}
```

if it is `nil`, we should return an error here.